### PR TITLE
KAN-6: Implemented `GET v1/api/host-me/countries/{countryId}/cities` and `GET v1/api/host-me/countries/{countryId}/cities/{cityId}` & removed custom JOOQ generator strategy

### PR DESCRIPTION
**Pull request by lilOlbee**

### CODE REVIEW

File: adapter/rest-api/src/main/java/com/di/hostme/service/controller/CityController.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+@RestController
+@RequiredArgsConstructor
+public class CityController extends BaseController {
+
+  public static final String PARAM_CITY_ID = "cityId";
+
+  public static final String CITIES_URI = COUNTRY_URI + "/cities";
+  public static final String CITY_URI = CITIES_URI + "/{" + PARAM_CITY_ID + "}";
+
+  private static final Logger log = LoggerFactory.getLogger(CityController.class);
```
Comment (br4vespirit) [inline adapter/rest-api/src/main/java/com/di/hostme/service/controller/CityController.java:41]: Add TODO for logger to use AOP in future

File: business/src/main/java/com/di/hostme/service/business/city/FindCityService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class FindCityService extends AbstractService implements FindCityUseCase {
+
+  private final QueryCityPort queryCityPort;
+
+  @Override
+  public CityEntity execute(FindCityByIdQuery query) {
+    log.info("Applying findCityUseCase, query: '{}'.", query);
```
Comment (br4vespirit) [inline business/src/main/java/com/di/hostme/service/business/city/FindCityService.java:19]: Add TODO to use AOP in future

File: business/src/main/java/com/di/hostme/service/business/city/ListCityService.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class ListCityService extends AbstractService implements ListCityUseCase {
+
+  private final QueryCityPort queryCityPort;
+
+  @Override
+  public List<CityEntity> execute() {
+    log.info("Applying listCityUseCase.");
```
Comment (br4vespirit) [inline business/src/main/java/com/di/hostme/service/business/city/ListCityService.java:19]: Add TODO to use AOP in future

**Review by br4vespirit — COMMENTED** [review]
Add TODOs to not forget in future

File: business/src/main/java/com/di/hostme/service/business/city/ListCityService.java (hunk 3 above)
Comment (lilOlbee) [reply business/src/main/java/com/di/hostme/service/business/city/ListCityService.java:20]: Done

File: business/src/main/java/com/di/hostme/service/business/city/FindCityService.java (hunk 2 above)
Comment (lilOlbee) [reply business/src/main/java/com/di/hostme/service/business/city/FindCityService.java:20]: Done

File: adapter/rest-api/src/main/java/com/di/hostme/service/controller/CityController.java (hunk 1 above)
Comment (lilOlbee) [reply adapter/rest-api/src/main/java/com/di/hostme/service/controller/CityController.java:42]: Done

**Review by br4vespirit — APPROVED** [review]

**Pull request merged**